src/plugins/intel_npu/src/backend/src/zero_executor.cpp

```diff
@@ -63,42 +63,59 @@ ZeroExecutor::ZeroExecutor(const std::shared_ptr<const ZeroInitStructsHolder>& i
                                      _initStructs->getCommandQueueDdiTable(),
                                      _config,
                                      _group_ordinal);
-    _logger.debug("ZeroExecutor::ZeroExecutor - create fence");
-    Fence fence(graph_command_queue, _config);
 
-    _logger.debug("ZeroExecutor::ZeroExecutor - create graph");
-    OV_ITT_TASK_CHAIN(ZERO_EXECUTOR_GRAPH, itt::domains::LevelZeroBackend, "Executor::ZeroExecutor", "graphCreate");
-
-    ze_graph_desc_t desc{ZE_STRUCTURE_TYPE_GRAPH_DESC_PROPERTIES,
-                         nullptr,
-                         ZE_GRAPH_FORMAT_NATIVE,
-                         _networkDesc->compiledNetwork.size(),
-                         _networkDesc->compiledNetwork.data(),
-                         nullptr};
-    zeroUtils::throwOnFail(
-        "pfnCreate",
-        _graph_ddi_table_ext->pfnCreate(_initStructs->getContext(), _initStructs->getDevice(), &desc, &_graph));
-
-    OV_ITT_TASK_NEXT(ZERO_EXECUTOR_GRAPH, "pfnGetProperties");
-    zeroUtils::throwOnFail("pfnGetProperties", _graph_ddi_table_ext->pfnGetProperties(_graph, &_props));
-
-    auto targetDriverExtVersion = _initStructs->getDriverExtVersion();
-    if (targetDriverExtVersion <= ZE_GRAPH_EXT_VERSION_1_1) {
-        OPENVINO_THROW("Incompatibility between the NPU plugin and driver! The driver version is too old, please "
-                       "update the driver version");
-    }
+    // _graph is a nullptr for CIP path
+    // _graph will get graphHandle from compiler for CiD path
+    if (_networkDesc->graphHandle == nullptr) {
```

Comment:
We need such a check for pfnDestroy here as well. Othewise we might try to destroy it twice in case of CiD?

Reply:
nvmd, you destroy it now only if the executor is not created.